app/models/mixins/authentication_mixin.rb

```diff
@@ -158,6 +158,8 @@ def update_authentication(data, options = {})
       cred.userid = value[:userid]
       cred.password = value[:password]
       cred.auth_key = value[:auth_key]
+      # TODO(lwander) this should have its own cred attribute
+      cred.password = value[:json] unless value[:json].blank?
```

Comment:
I'd rather add the new DB field, but I'm not sure how to go about it. Could you point me to the code that does for `:password`?